mono/utils/mono-hwcap-arm.c

```diff
@@ -150,6 +157,23 @@ mono_hwcap_arch_init (void)
 			}
 		}
 
+		/*
+		 * If we're executing as a 32-bit binary on a 64-bit system, we can
+		 * safely assume the presence of all 32-bit features.
+		 */
+		if (arm64) {
+			mono_hwcap_arm_is_v5 = TRUE;
+			mono_hwcap_arm_is_v6 = TRUE;
+			mono_hwcap_arm_is_v7 = TRUE;
+
+			mono_hwcap_arm_has_vfp = TRUE;
+			mono_hwcap_arm_has_vfp3 = TRUE;
+			mono_hwcap_arm_has_vfp3_d16 = TRUE;
+
+			mono_hwcap_arm_has_thumb = TRUE;
+			mono_hwcap_arm_has_thumb2 = TRUE;
+		}
```

Comment:
OK, digging through some ARM docs, it seems Thumb and Thumb-2 are both supported in ARMv8. See here: http://www.arm.com/products/processors/armv8-architecture.php